src/core/model_config.proto

```diff
@@ -327,6 +327,14 @@ message ModelOutput
   //@@     model.
   //@@
   bool is_shape_tensor = 6;
+
+  //@@  .. cpp:var:: string scatter_with_input_shape
+  //@@
+  //@@     This field can be specified if the output  should be scattered
+  //@@     according to a the input's shape. The dynamic dimension of the
+  //@@     output will be set to the value of the same dimension in the input.
+  //@@
+  string scatter_with_input_shape = 7;
```

Comment:
I would rather have this handled in a manner similar to batch_input. So Add a batch_output with structure similar to batch_input.

Reply:
Changed the option to be a BatchOutput message